fedimint-bin-tests/src/util.rs

```diff
@@ -14,17 +14,42 @@ use super::*;
 #[derive(Debug, Clone)]
 pub struct ProcessHandle(Arc<ProcessHandleInner>);
 
+impl ProcessHandle {
+    pub async fn kill_process(self) -> Result<()> {
+        let ProcessHandle(arc_process_handle_inner) = self;
+        let process_handle_inner = &mut Arc::try_unwrap(arc_process_handle_inner).unwrap();
+        let mut child = std::mem::take(&mut process_handle_inner.child)
+            .ok_or_else(|| anyhow!("Unable to own ProcessHandleInner"))?;
+        info!("killing {}", process_handle_inner.name.clone());
+        let _ = nix::sys::signal::kill(
+            nix::unistd::Pid::from_raw(child.id().expect("pid should be present") as _),
+            nix::sys::signal::Signal::SIGTERM,
+        );
+        child.wait().await?;
+        Ok(())
+    }
+}
+
 #[derive(Debug)]
 pub struct ProcessHandleInner {
     name: String,
-    child: Child,
+    child: Option<Child>,
```

Comment:
not needed, you can move out of the struct in fn kill() to avoid the drop being called.

Reply:
I think it is needed. As we are taking out `Fedimintd` from members: `Vec<Option<Fedimintd>>` inside the `kill-server()` method, and so it's value will eventually drop, and `ProcessHandleInner::Drop` will be called and if there is no check, it will try to kill the process again.
Am I missing something?

Reply:
look up rust partial struct moves.

Reply:
update: this was just wrong, you can't move out of structs that implement drop :facepalm: